bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqBinding.java

```diff
@@ -0,0 +1,374 @@
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+
+		if (pollFuture != null && !pollFuture.isCancelled())
+			pollFuture.cancel(true);
+
+		if (pollResetFuture != null && !pollResetFuture.isCancelled())
+			pollResetFuture.cancel(true);
+	}
+
+	/**
+	 * Poll for device changes
+	 */
+	private void poll() {
+		if (invalidCredentials || this.myqOnlineData == null) {
+			logger.debug("Invalid Account Credentials");
+			return;
+		}
+
+		try {
+			// Get myQ Data
+			GarageDoorData garageStatus = myqOnlineData.getGarageData();
+
+			for (MyqBindingProvider provider : providers) {
+				for (String mygItemName : provider.getInBindingItemNames()) {
+					MyqBindingConfig deviceConfig = getConfigForItemName(mygItemName);
+
+					if (deviceConfig != null) {
+						GarageDoorDevice garageopener = garageStatus
+								.getDevice(deviceConfig.deviceIndex);
+						if (garageopener != null) {
+							if (deviceConfig.type instanceof StringItem) {
+								eventPublisher.postUpdate(mygItemName,
+										new StringType(garageopener.getStatus()
+												.getLabel()));
+							}
+							if (deviceConfig.type instanceof ContactItem) {
+								if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+									eventPublisher.postUpdate(mygItemName,
+											OpenClosedType.CLOSED);
+								} else {
+									eventPublisher.postUpdate(mygItemName,
+											OpenClosedType.OPEN);
+								}
+							}
+							if (deviceConfig.type instanceof RollershutterItem) {
+								if (garageopener.getStatus() == GarageDoorStatus.CLOSED) {
+									eventPublisher.postUpdate(mygItemName,
+											UpDownType.DOWN);
+								} else if (garageopener.getStatus() == GarageDoorStatus.OPEN) {
+									eventPublisher.postUpdate(mygItemName,
+											UpDownType.UP);
+								}
+								// if its not open, close or unknown then its in
+								// a half-way state.
+								else if (garageopener.getStatus() != GarageDoorStatus.UNKNOWN) {
+									eventPublisher.postUpdate(mygItemName,
+											new PercentType(50));
+								}
+							}
+
+							// make sure we are polling frequently
+							if (garageopener.getStatus().inMotion()) {
+								beginRapidPoll(false);
+							}
+						}
+					}
+				}
+			}
+		} catch (InvalidLoginException e) {
+			logger.error("Could not log in, please check your credentials.", e);
+			invalidCredentials = true;
+		} catch (IOException e) {
+			logger.error("Could not connect to MyQ service", e);
+		}
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void internalReceiveCommand(String itemName, Command command) {
+		super.internalReceiveCommand(itemName, command);
+		logger.debug("MyQ binding received command '{}' for item '{}'",
+				command, itemName);
+		if (myqOnlineData != null) {
+			computeCommandForItem(command, itemName);
+		}
```

Comment:
Might want to log a WARN if you are going to drop the command due to no `myqOnlineData`.